### ISSUE TITLE
[Service Bus] Emulator connection string support

### ISSUE THREAD

**Issue by jsquire** [Azure/azure-sdk-for-python#34273]
# Summary

The Azure Messaging team is working on a service emulator intended to provide a local development experience.  To reduce friction for developers, the emulator will be using connections without TLS so that certificates do not need to be managed.   To support this flow, the emulator will emit a connection string that contains an identifying slug to indicate to clients that they should not use TLS and should restrict communication to the local environment.

Emulator connection string slug:

```text
;UseDevelopmentEmulator=true
```

Full example:
```text
Endpoint=localhost:6765;SharedAccessKeyName=<< REDACTED >>;SharedAccessKey=<< REDACTED >>;UseDevelopmentEmulator=true
```

## Scope of work

- Update connection string parsing to recognize the emulator slug.
- When the emulator slug is present in a connection string, validate that the endpoint resolves to the local host.
- When the emulator slug is present in a connection string, do not use TLS for service operations.
- Do not allow non-TLS communication for non-local hosts.

## Out of scope

- Support for validating that custom host names (via local DNS or hosts file) resolve to localhost

## Success Criteria

- Connection string parsing recognizes the emulator slug.
- Connection string parsing preserves a custom port for the endpoint.
- When the emulator slug is present, connection string parsing validates that the endpoint resolves to local host.
- Local host validation allows for `localhost`, IPv4, and IPv6 forms.
- Clients support non-TLS communication for service operations.
- Clients require TLS communication for non-local hosts.
- New tests necessary to validate the feature have been created and pass reliably.
- The existing test suite continues to produce deterministic results and pass reliably.

## References and related

- [Azure Feedback feature tracking](https://feedback.azure.com/d365community/idea/d45a6423-2225-ec11-b6e6-000d3a4f0f84)

**Comment by jsquire** [comment]
//cc: @kashifkhan, @l0lawrence, @swathipil

**Comment by ColinBe95** [comment]
Hey @jsquire,
do you know if there is any updates to this? The service bus emulator is now publicly available, but without connection string support it remains unusable for python developers unfortunately. Would love to use it :)
Thank you so much!

**Comment by kashifkhan** [comment]
There is still work pending on it, but as soon as we have it out Ill update this thread.

**Comment by jvanegmond** [comment]
For anyone as incredibly curious and eager as I have been, you can already get it working with latest commits on this repo. Alternatively, you can take the currently released version and disable TLS with a workaround:

```
import asyncio

from azure.servicebus import ServiceBusMessage, TransportType
from azure.servicebus._pyamqp import AMQPClient
from azure.servicebus.aio import (
    ServiceBusClient,
)

# Disable TLS. Workaround for https://github.com/Azure/azure-sdk-for-python/issues/34273
org_init = AMQPClient.__init__
def new_init(self, hostname, **kwargs):
    kwargs["use_tls"] = False
    org_init(self, hostname, **kwargs)
AMQPClient.__init__ = new_init


async def main() -> None:
    queue_name = "queue.1"

    async with ServiceBusClient.from_connection_string(
        conn_str="Endpoint=sb://servicebus-emulator:5672;SharedAccessKeyName=1234;SharedAccessKey=password;UseDevelopmentEmulator=true",
        transport_type=TransportType.Amqp,
        retry_total=0,
    ) as client:
        async with client.get_queue_sender(queue_name) as sender:
            single_message = ServiceBusMessage("Hello, world!")
            await sender.send_messages(single_message)

        async with client.get_queue_receiver(queue_name, max_wait_time=30) as receiver:
            async for msg in receiver:
                print(str(msg))


asyncio.run(main())
```

**Comment by jvanegmond** [comment]
@jsquire Could these changes please be wrapped up and released? I have a PR on my side that is blocked on these changes being released.

**Comment by l0lawrence** [comment]
Hi @jvanegmond these changes will be released on our monthly release cadence -- I will update this issue when they are out :)

**Comment by chandni-enable** [comment]
@l0lawrence any approximate timeline of release? my team and I are also blocked with this issue and would really appreciate a release for this fix soon

**Comment by l0lawrence** [comment]
Hi @chandni-enable, eta is early February

**Comment by ColinBe95** [comment]
I believe with release [7.14.0](https://github.com/Azure/azure-sdk-for-python/releases/tag/azure-servicebus_7.14.0) this issue can be closed

**Comment by kashifkhan** [comment]
@ColinBe95 yes that is correct :) 

@jvanegmond @chandni-enable and others, the latest release on pypi has emulator support
